parcels/particleset.py

```diff
@@ -919,197 +753,82 @@ def execute(
                 self._kernel = pyfunc
             else:
                 self._kernel = self.Kernel(pyfunc)
+
         if output_file:
             output_file.metadata["parcels_kernels"] = self._kernel.name
 
-        # Set up the interaction kernel(s) if not set and given.
-        if self._interaction_kernel is None and pyfunc_inter is not None:
-            if isinstance(pyfunc_inter, InteractionKernel):
-                self._interaction_kernel = pyfunc_inter
+        # The fieldset time intervale defines the extent of time that is allowed to be
+        # simulated. If `fieldset.time_interval` is not None, it will be used to determine the endtime (the min of endtime or fieldset.time_interval[1]).
+        # If `fieldset.time_interval` is None, the endtime will be determined by the
+        # `endtime` parameter or the fieldset's time dimension.
+        # Time parameters for the main for loop are converted to floats, since the interpolation kernels expect float objects for time
```

Comment:
Ah, this is an interesting inning. Should we change that too, in the long run? Keeping timedelta64 throughout may avoid round-off issues like in #2045

Reply:
The time handling in the pset.execute function has been fully converted to datetime or timedelta objects and I've brought back in the `runtime` argument, per the other discussions. As a result, this comment is no longer entirely relevant